pi-manage

```diff
@@ -1319,6 +1284,252 @@ def import_tokens(file, tokenrealm=None):
+            if exists:
+                if not update:
+                    print("{0!s} {1!s} exists and -u is not specified, "
+                          "skipping import.".format(conftype, name))
+                    continue
+                else:
+                    action_str = "Updated"
+            if conftype == "policy":
+                r = set_policy(name,
+                               action=config.get("action"),
+                               active=config.get("active", True),
+                               adminrealm=config.get("adminrealm"),
+                               adminuser=config.get("adminuser"),
+                               check_all_resolvers=config.get(
+                                   "check_all_resolvers", False),
+                               client=config.get("client"),
+                               # condition=policy.get("condition"),
+                               realm=config.get("realm"),
+                               resolver=config.get("resolver"),
+                               scope=config.get("scope"),
+                               time=config.get("time"),
+                               user=config.get("user"))
+            elif conftype == "resolver":
+                resolvertype = config.get("type")
+                data = config.get("data")
+                # now we can create the resolver
+                params = {'resolver': name, 'type': resolvertype}
+                for key in data.keys():
+                    params.update({key: data.get(key)})
+                r = save_resolver(params)
+            elif conftype == "event":
+                r = set_event(name, config.get("event"), config.get("handlermodule"),
+                              config.get("action"),
+                              conditions=config.get("conditions"),
+                              ordering=config.get("ordering"),
+                              options=config.get("options"),
+                              active=config.get("active"),
+                              position=config.get("position", "post"),
+                              id=event_id)
+
+            print("{0!s} {1!s} {2!s} with result {3!s}".format(action_str, conftype, name, r))
+
+
+@config_export_manager.option("--print_passwords", "-p", action="store_true",
+                              help="Print the passwords used in the resolver configuration. "
+                                   "This will overwrite existing passwords on import.")
+@config_export_manager.option("--archive", "-a", action="store_true",
+                              help="Compress the created config-backup as tar.gz archive instead "
+                                   "of printing to standard out.")
+@config_export_manager.option("--directory", "-d", action="store_true",
+                              help="Directory where the backup will be stored.")
+def fullexport(directory=None, archive=False, print_passwords=False):
```

Comment:
We could add the command class explicitly (without the decorators):
https://flask-script.readthedocs.io/en/latest/#adding-arguments-to-commands

Reply:
Done the refactoring in https://github.com/privacyidea/privacyidea/pull/2468/commits/355c90262aefa5132729c53b1de338265b7e80d9